File: CurrencyConverter.cpp
/*Simple Currency Converter in C++, with 4 
 currencies: EURO, US Dollar, British Pound and Chinese Yen. 
 And a function to limit user input*/


#include <iostream>
using namespace std;

void letterInput(char want1, char want2, char want3, char want4, char &input);

int main ()
{
    
    //Header and menu asking for starting currency
    cout << "\t\tCurrency Converter \n\nWhich currency do you want to exchange ?";
    cout << "\nEUR (a)\t\tUSD (b)\t\tGBP (c)\t\tCNY (d)\n";                                             
    char input, output;
    string icurrency, ocurrency;
    cin >> input;                                                                               
    letterInput ('a','b','c','d', input); //Function to limt userinput to "abcd" ONLY
    
    //Switch statement to turn single letter user input into actual currency names
    switch (input)
    {
        case 'a':
        icurrency = "EUR";
        break;

        case 'b':
        icurrency = "USD";
        break;

        case 'c':
        icurrency = "GBP";
        break;

        case 'd':
        icurrency = "CNY";
        break;
    }

    //Menu for taking the currency amount and the target currency
    cout << "\nHow much " << icurrency << " do you have ?\n";                                  
    float multiplyer, result, amount;
    cin >> amount;
    cout << "\nWhich currency do you want instead ?\n";
    cout << "EUR (a)\t\tUSD (b)\t\tGBP (c)\t\tCNY (d)\n";
    cin >> output;
    letterInput ('a','b','c','d', output); //Function for limiting user input to "abcd" only

    //Switch statement for turning single letter input into actual currency names and setting 
    //the multiplyer to the needed exchange rate 
    switch (output)
    {
        case 'a':
        ocurrency = "EUR";
        switch (input)
        {
            case 'a':
            multiplyer = 1;
            break;

            case 'b':
            multiplyer = 1.1;
            break;

            case 'c':
            multiplyer = 0.83;
            break;

            case 'd':
            multiplyer = 7;
            break;
        }
        break;

        case 'b':
        ocurrency = "USD";
        switch (input)
        {
            case 'a':
            multiplyer = 0.91;
            break;

            case 'b':
            multiplyer = 1;
            break;

            case 'c':
            multiplyer = 0.76;
            break;

            case 'd':
            multiplyer = 6.37;
            break;
        }
        
        break;

        case 'c':
        ocurrency = "GBP";
        switch (input)
        {
            case 'a':
            multiplyer = 1.2;
            break;

            case 'b':
            multiplyer = 1.32;
            break;

            case 'c':
            multiplyer = 1;
            break;

            case 'd':
            multiplyer = 8.39;
            break;
        }
        break;

        case 'd':
        ocurrency = "CNY";
        switch (input)
        {
            case 'a':
            multiplyer = 0.143;
            break;

            case 'b':
            multiplyer = 0.157;
            break;

            case 'c':
            multiplyer = 0.12;
            break;

            case 'd': 
            multiplyer = 1;
            break;

        }
        break;

    }
    result = amount * multiplyer; //Result = amount * exchange rate
    cout << "\nYour " << amount << " " << icurrency << " are " << result << " " << ocurrency; 

    return 0;
}

//Function that limits user input to 4 single letters 
void letterInput(char want1, char want2, char want3, char want4, char &input) 
{
    if (input == want1 || input == want2 || input == want3 || input == want4)
    {
        return;
    }
    else 
    {
        printf("\nWrong input, try again !");
        printf("\nEUR (a)\t\tUSD (b)\t\tGBP (c)\t\tCNY (d)\n");
        cin >> input;    
        letterInput (want1, want2, want3, want4, input);
    }
}

File: CurrencyConverter.java
//Simple Currency Converter in JAVA with limited user input

import java.util.*;


public class CurrencyConverter 
{
    public static void main (String [] args)
    {
        //Input Currency menu
        System.out.println("*****Currency Converter in JAVA*****\n\n");
        System.out.print("Which Currency do you want to convert ?\nEUR (a)   USD (b)   GBP (c)   CNY (d)\nChoice:  ");
        Scanner input = new Scanner (System.in).useLocale(Locale.US);
        String iCurrency = input.next();

        //While loop to limit user input to abcd 
        while (!iCurrency.equals ("a") && !iCurrency.equals ("b") && !iCurrency.equals ("c") && !iCurrency.equals ("d"))
        {
                System.out.print("\nWrong input, try again !\nEUR (a)  USD (b)  GBP (c)  CNY (d)\nChoice:  ");
                iCurrency = input.next();
        }
        String iCurrencyName = "0";

        //Turn single letter input into a real currency name
        switch (iCurrency)
        {
            case "a":
            iCurrencyName = "EUR";
            break;

            case "b":
            iCurrencyName = "USD";
            break;

            case "c":
            iCurrencyName = "GBP";
            break;

            case "d":
            iCurrencyName = "CNY";
            break;
        }
        
        //Currency amount menu 
        System.out.print("\nHow much " + iCurrencyName + " do you have ?: ");
        double amount = input.nextDouble();

        //Target currency menu 
        System.out.print("\nWhich Currency do you want instead ?\nEUR (a)   USD (b)   GBP (c)   CNY (d)\nChoice:  ");
        String oCurrency = input.next();

        //while loop to limit user input to abcd only 
        while (!oCurrency.equals ("a") && !oCurrency.equals ("b") && !oCurrency.equals ("c") && !oCurrency.equals ("d"))
        {
                System.out.print("\nWrong input, try again !\nEUR (a)  USD (b)  GBP (c)  CNY (d)\nChoice:  ");
                oCurrency = input.next();
        }

        String oCurrencyName = "0";   
        double multiplyer = 0; 
        
        //Turn single letter input into a real currency name and choose the right exchange rate
        switch (oCurrency)
        {
            case "a":
            oCurrencyName = "EUR";
            switch (iCurrency)
            {
                case "a":
                multiplyer = 1.0;
                break;
                case "b":
                multiplyer = 0.91;
                break;
                case "c":
                multiplyer = 1.19;
                break;
                case "d":
                multiplyer = 0.14;
                break;  
            }            
            break;

            case "b":
            oCurrencyName = "USD";
            switch (iCurrency)
            {
                case "a":
                multiplyer = 1.1;
                break;
                case "b":
                multiplyer = 1;
                break;
                case "c":
                multiplyer = 1.32;
                break;
                case "d":
                multiplyer = 0.16;
                break;  
            }            
            break;

            case "c":
            oCurrencyName = "GBP";
            switch (oCurrency)
            {
                case "a":
                multiplyer = 0.84;
                break;
                case "b":
                multiplyer = 0.76;
                break;
                case "c":
                multiplyer = 1;
                break;
                case "d":
                multiplyer = 0.12;
                break;  
            }            
            break;

            case "d":
            oCurrencyName = "CNY";
            switch (iCurrency)
            {
                case "a":
                multiplyer = 7.01;
                break;
                case "b":
                multiplyer = 6.35;
                break;
                case "c":
                multiplyer = 8.38;
                break;
                case "d":
                multiplyer = 1;
                break;  
            }            
            break;
        }     

        double result = amount * multiplyer;

        //Output result
        System.out.print("\nYour " + amount + " " + iCurrencyName  + " are " + result + " " + oCurrencyName);
        
        input.close();
    }

}
